haystack/nodes/base.py

```diff
@@ -74,6 +70,38 @@ def __init_subclass__(cls, **kwargs):
         # Enables generic load() for all specific component implementations.
         cls._subclasses[cls.__name__] = cls
 
+    @property
+    def name(self) -> Optional[str]:
+        if "name" not in self._component_config:
+            return None
+        return self._component_config["name"]
```

Comment:
Just for style, this can become a oneliner: `return self._component_config.get("name", None)`

Reply:
Yes, I totally forgot about dict.get :D